### PR TITLE
WIP B3 Reports (biyeun to self review)

### PR DESCRIPTION
**Pull request by biyeun**

### CODE REVIEW

File: corehq/apps/reports/dont_use/fields.py [hunk 1]
```diff
@@ -45,10 +53,16 @@ def update_context(self):
         """
         pass
 
+    def get_bootstrap_template(self):
+        if self.is_bootstrap3:
+            return self.template.replace('/bootstrap2/','/bootstrap3/')
```
Comment (dimagimon) [inline corehq/apps/reports/dont_use/fields.py]: E231 missing whitespace after ','

File: corehq/apps/reports/filters/base.py [hunk 2]
```diff
@@ -65,12 +75,17 @@ def render(self):
...
-        return render_to_string(self.template, self.context)
+        return render_to_string(self.get_bootstrap_template(), self.context)
 
     @classmethod
     def get_value(cls, request, domain):
         return request.GET.get(cls.slug)
 
+    def get_bootstrap_template(self):
+        if self.is_bootstrap3:
+            return self.template.replace('/bootstrap2/','/bootstrap3/')
```
Comment (dimagimon) [inline corehq/apps/reports/filters/base.py]: E231 missing whitespace after ','

File: corehq/apps/reports/generic.py [hunk 3]
```diff
@@ -247,36 +249,44 @@ def timezone(self):
...
     def template_report_partial(self):
-        return self.report_partial_path
+        return self._select_bootstrap_template(self.report_partial_path)
 
     @property
     @memoized
     def template_filters(self):
-        return self.base_template_filters or "reports/async/bootstrap2/filters.html"
+        return self._select_bootstrap_template(
+           self.base_template_filters or "reports/async/bootstrap2/filters.html"
```
Comment (dimagimon) [inline corehq/apps/reports/generic.py]: E121 continuation line under-indented for hanging indent

File: corehq/apps/hqadmin/reports.py [hunk 4]
```diff
@@ -534,6 +534,8 @@
 
 class AdminReport(GenericTabularReport):
```
Comment (biyeun) [inline corehq/apps/hqadmin/reports.py]: done

File: corehq/apps/hqwebapp/static/hqwebapp/js-custom/hq.legacy.js [hunk 5]
```diff
@@ -15,10 +15,9 @@ var COMMCAREHQ = (function () {
             var el = $(
```
Comment (biyeun) [inline corehq/apps/hqwebapp/static/hqwebapp/js-custom/hq.legacy.js]: done

File: corehq/apps/reports/cache.py [hunk 6]
```diff
@@ -13,6 +13,7 @@ class _ReportQuickCache(QuickCache):
     """
     def __call__(self, *args, **kwargs):
         report = args[0]
+        self.prefix = '{}{}'.format(self.prefix, '/b3' if report.is_bootstrap3 else '')
```
Comment (biyeun) [inline corehq/apps/reports/cache.py]: done

File: corehq/apps/reports/templates/reports/dont_use_fields/bootstrap3/multiselect_generic.html [hunk 7]
```diff
@@ -0,0 +1,38 @@
+{% load hq_shared_tags %}
```
Comment (biyeun) [inline corehq/apps/reports/templates/reports/dont_use_fields/bootstrap3/multiselect_generic.html]: done

File: corehq/apps/reports/templates/reports/dont_use_fields/bootstrap3/select_generic.html [hunk 8]
```diff
@@ -0,0 +1,32 @@
+{% load hq_shared_tags %}
```
Comment (biyeun) [inline corehq/apps/reports/templates/reports/dont_use_fields/bootstrap3/select_generic.html]: done

File: corehq/apps/reports/dont_use/fields.py [hunk 9]
```diff
@@ -12,6 +12,10 @@
 from django.utils.translation import ugettext_noop
 from django.utils.translation import ugettext as _
 import uuid
+from corehq.apps.reports.util import (
```
Comment (biyeun) [inline corehq/apps/reports/dont_use/fields.py]: done

File: corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js [hunk 10]
```diff
@@ -99,25 +99,46 @@ function HQReportDataTables(options) {
                 params.bProcessing = true;
                 params.sAjaxSource = self.ajaxSource;
                 params.bFilter = $(this).data('filter') || false;
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/javascripts/config.dataTables.bootstrap.js]: done

File: corehq/apps/reports/static/reports/javascripts/reports.async.js [hunk 11]
```diff
@@ -187,13 +187,16 @@ var HQAsyncReport = function (o) {
         self.updateReport(true, window.location.search.substr(1));
     });
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/javascripts/reports.async.js]: done

File: corehq/apps/reports/static/reports/javascripts/reports.config.js [hunk 12]
```diff
@@ -130,17 +130,23 @@ var HQReport = function (options) {
             $(self.toggleFiltersButton).button('open');
```
Comment (biyeun) [inline corehq/apps/reports/static/reports/javascripts/reports.config.js]: done

**Pull request closed without merging**